dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -359,20 +389,47 @@ public String toJson(final Contentlet contentlet) throws DotMappingException {
                 mlowered.put(lowerCaseKey, lowerCaseValue);
 			}
 
-			final Optional<Field> binaryField = this.findFirstBinaryFieldIndexable(contentlet);
-			if (binaryField.isPresent()) {
+			if (Config.getBooleanProperty(WRITE_METADATA_ON_REINDEX, true)) {
 
-				this.generateBinaryMetadata(contentlet, sw, mlowered, binaryField.get());
+				if (contentlet.isLive() || contentlet.isWorking()) {
```

Comment:
I think this validation is not required because a content will always be live and/or working. It will always evaluate true

Reply:
right

Reply:
yeah it's coming from the old logic that was already in place. it's just a refactoring

Reply:
okey dokey - removed